tests/fixtures.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common fixtures."""
+
+import collections
+import itertools
+import os
+import shutil
+import tempfile
+from pathlib import Path
+
+import pytest
+import tftest
+import yaml
+
+PlanSummary = collections.namedtuple('PlanSummary', 'values counts outputs')
+
+
+def plan_summary(module_path, basedir, tf_var_files=None, **tf_vars):
+  """
+  Run a Terraform plan on the module located at `module_path`.
+
+  - module_path: terraform root module to run. Can be an absolute
+    path or relative to the root of the repository
+
+  - basedir: directory root to use for relative paths in
+    tf_var_files.
+
+  - tf_var_files: set of terraform variable files (tfvars) to pass
+    in to terraform
+
+  Returns a PlanSummary object containing 3 attributes:
+  - values: dictionary where the keys are terraform plan addresses
+    and values are the JSON representation (converted to python
+    types) of the attribute values of the resource.
+
+  - counts: dictionary where the keys are the terraform resource
+    types and the values are the number of times that type appears
+    in the plan
+
+  - outputs: dictionary of the modules outputs that can be
+    determined at plan type.
+
+  Consult [1] for mode details on the structure of values and outputs
+
+  [1] https://developer.hashicorp.com/terraform/internals/json-format
+  """
+  module_path = Path(__file__).parents[1] / module_path
+
+  # FIXME: find a way to prevent the temp dir if TFTEST_COPY is not
+  # in the environment
+  with tempfile.TemporaryDirectory(dir=module_path.parent) as tmp_path:
+    # if TFTEST_COPY is set, copy the fixture to a temporary
+    # directory before running the plan. This is needed if you want
+    # to run multiple tests for the same module in parallel
+    if os.environ.get('TFTEST_COPY'):
+      test_path = Path(tmp_path)
+      shutil.copytree(module_path, test_path, dirs_exist_ok=True)
```

Comment:
nit: what about using `ignore` in `copytree()` instead of copying and deleting after?

Reply:
Beautiful. Done